### DISCUSSION TITLE
kombu.exceptions.EncodeError

### DISCUSSION THREAD

**Discussion by geeeez** [celery/celery#7665] [Q&A]
error->kombu.exceptions.EncodeError: Object of type Case is not JSON serializable
<img width="636" alt="image" src="https://user-images.githubusercontent.com/29922543/172174948-f1e3f1c2-55cf-4171-807d-b8b8fc8c0190.png">
celery.py
```python
from celery import Celery
cel = Celery('celery_demo',
             broker='redis://127.0.0.1:6379/1',
             backend='redis://127.0.0.1:6379/2',
             # accept_content=['pickle', 'json'],
             # task_serializer='pickle',
             # result_serializer="pickle",
             # 包含以下两个任务文件，去相应的py文件中找任务，对多个任务做分类
             include=['celery_dispatch.celery_task_casecall'
                      ])
# 时区
cel.conf.timezone = 'Asia/Shanghai'
# 是否使用UTC
cel.conf.enable_utc = False
```

task.py

```python

@cel.task
def case_call(case):
    case.call()
```
The case is a class. It will report an error when I want to pass it as a parameter。
How can I pass a custom class object as a parameter and run normally?

[a comment by an automated account is not shown]

**Comment by orireiter** [comment]
hi @geeeez,

To my understanding, the problem is not with Celery or Kombu.
Generally, custom classes in python are not JSON serializable, as seen in the pictures I attached, as well in yours.

<img width="431" alt="Screen Shot 2022-07-16 at 16 08 52" src="https://user-images.githubusercontent.com/60241083/179356232-b7c55035-3841-4f6a-bceb-3f6f39b55af6.png">
<img width="572" alt="Screen Shot 2022-07-16 at 16 10 59" src="https://user-images.githubusercontent.com/60241083/179356297-f2d32808-792b-4f70-a088-3f0117fd63d4.png">

you could either work around it by sending through celery the parameters that would initialize you class, and init it in the task itself:
```python

@cel.task
def case_call(*args_to_init_case, **kwargs_to_init_case):
    case = Case(*args_to_init_case, **kwargs_to_init_case)
    case.call()
```

or create your own JSON encoders/decoders as shown here:
https://stackoverflow.com/questions/21631878/celery-is-there-a-way-to-write-custom-json-encoder-decoder

Hope I helped! 
These are the solutions I could think of for your problem.
